src/test/java/uk/gov/hmcts/reform/civil/service/robotics/mapper/EventHistoryMapperTest.java

```diff
@@ -180,6 +182,49 @@ void setup() {
             when(featureToggleService.isRpaContinuousFeedEnabled()).thenReturn(true);
         }
 
+        @Test
+        void shouldReturnLatestTriggerEvent() {
+            CaseData caseData = CaseDataBuilder.builder().atStateClaimDetailsNotified().build();
+            Event claimIssuedEvent = Event.builder()
+                .eventSequence(1)
+                .eventCode("999")
+                .dateReceived(caseData.getIssueDate().atStartOfDay())
+                .eventDetailsText("Claim issued in CCD.")
+                .eventDetails(EventDetails.builder()
+                                  .miscText("Claim issued in CCD.")
+                                  .build())
+                .build();
+            Event claimNotifiedEvent = Event.builder()
+                .eventSequence(2)
+                .eventCode("999")
+                .dateReceived(caseData.getClaimNotificationDate())
+                .eventDetailsText("Claimant has notified defendant.")
+                .eventDetails(EventDetails.builder()
+                                  .miscText("Claimant has notified defendant.")
+                                  .build())
+                .build();
+
+            Event claimDetailsNotifiedEvent = Event.builder()
+                .eventSequence(3)
+                .eventCode("999")
+                .dateReceived(caseData.getClaimDetailsNotificationDate())
+                .eventDetailsText("Claim details notified.")
+                .eventDetails(EventDetails.builder()
+                                  .miscText("Claim details notified.")
+                                  .build())
+                .build();
+
+            var eventHistory = mapper.buildEvents(caseData);
+
+            assertThat(eventHistory).isNotNull();
+            assertThat(eventHistory)
+                .extracting("miscellaneous")
+                .asList()
+                .containsExactly(claimIssuedEvent, claimNotifiedEvent, claimDetailsNotifiedEvent);
+            String triggerReason = findLatestEventTriggerReason(eventHistory);
+            assertTrue(triggerReason.matches("Claim details notified."));
+        }
+
```

Comment:
this was not required for this story as mapper update is removed but it is ok to leave it as it is just  testing extra functionality